ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -193,10 +194,10 @@ def start_learning(self, env, trainer_config):
 
         # Prevent a single session from taking all GPU memory.
         self.initialize_trainers(trainer_config)
+        policies = {brain_name: trainer.policy for brain_name, trainer in self.trainers.items()}
```

Comment:
@vincentpierre you mentioned some concerns about this, could you elaborate?  @eshvk is reviewer on this PR but wanted to make sure to get your input

Reply:
I'd be curious to know if this change still makes sense. We are now exposing to the `TrainerController` the `Policy` object(s). Initially, I think it made sense because the `Policy`(s) will be distributed to different `ActorManager` processes. How about now that we might not be doing that?

Reply:
I'd be also curious to hear your thoughts on how we could make the following a bit more explicit: Now, we have a local variable in `start_learning`,  `policies` that is mutated when `update_policy` is called in another method `take_step` for each iteration of the loop.

Reply:
I'd propose for now we just get rid of `TrainerController` knowing anything about policies until the time we need it.  I do think that the Policy is the right place for `get_action` so we can just additionally expose that via the Trainer.  This will make sense if we want a trainer to potentially know about multiple policies in the future as well.

Reply:
I made this change, PTAL.

Reply:
👍